crates/doc/src/parser/mod.rs

```diff
@@ -348,5 +348,45 @@ mod tests {
         assert!(matches!(fallback.source, ParseSource::Function(_)));
     }
 
-    // TODO: test regular doc comments & natspec
+    #[test]
+    fn contract_with_doc_comments() {
+        let items = parse_source(
+            r"
+            pragma solidity ^0.8.19;
+            /// @name Test
+            ///  no tag
+            ///@notice    Cool contract    
+            ///   @  dev     This is not a dev tag 
+            /**
+             * @dev line one
+             *    line 2
+             */
+            contract Test {
+                /*** my function    
+                      i like whitespace    
+            */
+                function test() {}
+            }
+        ",
+        );
+
+        assert_eq!(items.len(), 1);
+
+        let contract = items.first().unwrap();
+        assert_eq!(contract.comments.len(), 2);
+        assert_eq!(
+            *contract.comments.first().unwrap(),
+            Comment::new(CommentTag::Notice, "Cool contract".to_owned())
+        );
+        assert_eq!(
+            *contract.comments.get(1).unwrap(),
+            Comment::new(CommentTag::Dev, "line one\nline 2".to_owned())
+        );
+
+        let function = contract.children.first().unwrap();
+        assert_eq!(
+            *function.comments.first().unwrap(),
+            Comment::new(CommentTag::Notice, "my function\ni like whitespace".to_owned())
+        );
```

Comment:
I don't know if this is expected but it passes.

Reply:
This is indeed unexpected, we will need to look into this

Reply:
On further local testing, it seems that the parser interprets comments with no tags as notice. This is actually inline with what the [natspec docs say](https://docs.soliditylang.org/en/latest/natspec-format.html#tags) that the compiler does

Reply:
>As a special case, if no tags are used then the Solidity compiler will interpret a `///` or `/**` comment in the same way as if it were tagged with `@notice`.

Ah perfect, thanks for looking into this!

We will need to keep this in mind when we refactor the `forge doc` to use Solar cc @grandizzy